src/test/java/edu/oakland/test/middleware01/MiddlewareGisInterfaceImplementationTests.java

```diff
@@ -0,0 +1,53 @@
+package edu.oakland.test.middleware01;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import edu.oakland.production.middleware01.MiddlewareGisProcessor;
+import edu.oakland.test.middleware01.MiddlewareGisProcessorStub;
+import edu.oakland.production.middleware01.MiddlewareGisInterface;
+import edu.oakland.production.middleware01.MiddlewareGisInterfaceImplementation;
+
+import edu.oakland.helper.admin.Satellite;
+import java.lang.IllegalArgumentException;
+import edu.oakland.helper.admin.LocationDataPoint;
+
+@DisplayName("Middleware GIS interface test")
```

Comment:
Really minor nitpick, but I recommend calling it this so that the logs are consistent across all teams.
```suggestion
@DisplayName("MiddlewareGisInterfaceImplementation Unit Tests")
```